### DISCUSSION TITLE
installing to the environment with `make install`

### DISCUSSION THREAD

**Discussion by looooo** [prefix-dev/pixi#1618] [Q&A]
I tried pixi here:
https://github.com/looooo/SMESH/blob/occt7.8.1/pixi.toml
Now I run into the issue of installing SMESH via `make install`. With conda-build it's possible to use the $PREFIX variable. But for pixi $PREFIX is not set. So I am wondering how to achive the installation of SMESH into the conda-env?

**Comment by looooo** [comment]
ok, this one I have figured out by using something like: 
```
configure = { cmd = ["cmake", 
                     "-S", ".", 
                     "-D", "CMAKE_INSTALL_PREFIX:FILEPATH=$CONDA_PREFIX",
                     "-B", "build"], depends-on = ["initialize", "prepare"]}
```

but I have a followup question:

When I try to build the tests for the same library I am running into this issue:
```
ld: warning: directory not found for option '-F/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/System/Library/Frameworks'
dyld[25410]: Library not loaded: @rpath/libNETGENPlugin.dylib
  Referenced from: <95219AEC-5643-360E-89E3-D32F0DF9CB6C> /Users/lo/projects/freecad/SMESH/test/build/test_StdMeshers
  Reason: tried: '/usr/local/lib/libNETGENPlugin.dylib' (no such file), '/usr/lib/libNETGENPlugin.dylib' (no such file, not in dyld cache)
```

and running the tests fails because these libraries are not found. So I guess the LD_LIBRARY_PATH is not setup correctly. But I am not sure how to setup this path with pixi the right way. Does anyone knows what conda-build does to make the linking work correctly with conda-envs?

**Reply by looooo** [reply]
this did the job:
```
"-D", "CMAKE_INSTALL_RPATH=$CONDA_PREFIX/lib",
```

**Comment by looooo** [comment]
next issue: there is a test case which runs a catch2 executeable. There is a segfault which would be nice to debug with gdb. But running `gdb pixi r test` does not work for me.

**Reply by looooo** [reply]
partially solved by adding a task which runs gdb.

**Comment by looooo** [comment]
next question:

is there a way to specify platform specific dependencies for a feature, like I tried here (but this is not working):
```
[feature.build.dependencies]
vtk = ">=9.3.1,<9.4"
occt = ">=7.8.1,<7.9"
libzlib = ">=1.3.1,<1.4"
libboost-devel = ">=1.85.0,<1.86"
cmake = ">=3.30.0,<3.31"
zlib = ">=1.3.1,<1.4"
catch2 = "<3.0"

[feature.build.win-64.dependencies]
pthreads-win32 = "*"
```

**Reply by ruben-arts** [reply]
Yes you're almost there! 

```toml
[feature.build.target.win-64.dependencies]
pthreads-win32 = "*"
```

Checkout the docs for more options: https://pixi.sh/latest/reference/project_configuration/#the-feature-table

**Comment by looooo** [comment]
another question:
Does it make sense to devide the environments into `build`, `host` and `run`like it's done for conda-build? If so what are the best strategies to activate multiple environments (eg. build and host)?

**Comment by looooo** [comment]
I have another question:
I have a task which should be performed only on unix systems. This task is required for another task. Is there a way to make the other task depend on the first task only for unix systems?
